pkg/server/sandbox_run.go

```diff
@@ -456,9 +456,21 @@ func (c *criService) generateSandboxContainerSpec(id string, config *runtime.Pod
 	return g.Config, nil
 }
 
-// setupSandboxFiles sets up necessary sandbox files including /dev/shm, /etc/hosts
-// and /etc/resolv.conf.
+// setupSandboxFiles sets up necessary sandbox files including /dev/shm, /etc/hosts,
+// /etc/resolv.conf and /etc/hostname.
 func (c *criService) setupSandboxFiles(id string, config *runtime.PodSandboxConfig) error {
+	sandboxEtcHostname := c.getSandboxHostname(id)
+	if config.GetHostname() == "" {
+		// copy host's /etc/hostname
+		if err := c.os.CopyFile(etcHostname, sandboxEtcHostname, 0644); err != nil {
```

Comment:
I thought kubelet was ensuring /etc/hostname?? Shouldn't this be an error case?

Reply:
Kubelet doesn't do that.

Reply:
sent you a txt message to chat over this for a min

Reply:
yeah I was reading the ensure hosts code in kublet.. got off track.. thx